### DISCUSSION TITLE
Step by Step Install

### DISCUSSION THREAD

**Discussion by lucamazzasette** [cloud-carbon-footprint/cloud-carbon-footprint#1128] [General]
I'm sorry for request but I'm Newbie.
I try to install this fantastic tool into Azure or GCP but I'm not able.Is 
Is it possible to have a step by step guide for installing and connecting?

Thanks in advance for your support

**Comment by ccasher** [comment]
Hi @lucamazzasette, thanks for reaching out and happy to hear you are looking to install the tool. Have you had a chance to check out our [guided install documentation](https://www.cloudcarbonfootprint.org/docs/getting-started#guided-install)? Please read it over and let us know if you have further questions.

**Reply by lucamazzasette** [reply]
Thanks for reply, Yes I read the documentation but I'm in the step before. I don't understood where must be install.
I suppose on the Virtual machine whit Linux ( is Ubuntu OK?) and this "fantastic Tool" must be install after install prerequisite:
1. Install Ubuntu
2. sudo apt update
3. sudo apt install git
4. sudo apt install nodejs
5. sudo apt install npm
6. sudo apt install curl
7. curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
8. echo "deb https://dl.yarnpkg.com/debian/ stable main" |
    sudo tee /etc/apt/sources.list.d/yarn.list
9. sudo apt install yarn
10. git clone --branch latest https://github.com/cloud-carbon-footprint/cloud-carbon-footprint.git
11. cd cloud-carbon-footprint
12.yarn install && yarn guided-install

but on the last step I have many error and nont complete the install.

Is it the simple method for Newbie to install this app into GCP or Azure or On-prem?

Thanks for your time

**Reply by 4upz** [reply]
Hi @lucamazzasette thanks for the detailed steps! It seems you are trying to run both installation commands at the same time. You only need to run `yarn install` on its own  or you can do `yarn guided-install` by itself.

The difference is that the guided install command does `yarn install` for you and also provides an easy way to configure your credentials to the needed variables in your API's  .env files. It will also be helpful if you share what exact errors your seeing so we can also troubleshoot the exact issue.

**Reply by lucamazzasette** [reply]
Thanks for the answer,
 it's not clear to me if I have to install a virtual server first or I can run the command "yarn install" directly from the Google Cloud Platform shell.
Could you tell me the steps starting from zero?

**Reply by lucamazzasette** [reply]
When I try install  app with command : 
npx @cloud-carbon-footprint/create-app

I recived this error :   
Building the app:
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
  executing     yarn install ✖
There was an error creating your App: Could not execute command yarn install
We are going to clean up, and then you can try again.

 Cleanup:
  remove        temporary directory ✔

Failed to create app!

and if i try instll local with clone:
git clone --branch latest https://github.com/cloud-carbon-footprint/cloud-carbon-footprint.git
cd cloud-carbon-footprint
 yarn install && yarn guided-install

### I recived : 
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
cc@CarbonFootprint:~/cloud-carbon-footprint$ sudo yarn install
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'install'
cc@CarbonFootprint:~/cloud-carbon-footprint$

Please help me.

**Reply by 4upz** [reply]
@lucamazzasette This seems to be an ubuntu-specific issue pertaining to Yarn. I think the advice in this [Stack Overflow thread](https://stackoverflow.com/questions/46013544/yarn-install-command-error-no-such-file-or-directory-install) may help. As far as the order of commands, your steps seem to be fine as far as installing the virtual server first, environment tools (i.e. git, node, etc), then creating the app.

Just be aware that based on the original steps you've posted, you don't need to clone the repository if you're using the `create-app` method. It will create the necessary files and directory for you. Other than that, resolving the yarn issue should fix what you're facing.

**Reply by 4upz** [reply]
So to recap the steps, here is what your steps should look like based on your environment:

1. Install Ubuntu
2. sudo apt update
3. sudo apt install git
4. sudo apt install nodejs
5. sudo apt install npm
6. sudo apt install curl
7. curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
8. echo "deb https://dl.yarnpkg.com/debian/ stable main" |
9. sudo tee /etc/apt/sources.list.d/yarn.list
10. sudo apt install yarn
11. npx @cloud-carbon-footprint/create-app
12. cd [name-of-app]
13. yarn install
14. yarn guided-install

For the `ERROR: [Errno 2] No such file or directory: 'install'` issue, try out this [Stack Overflow thread](https://stackoverflow.com/questions/46013544/yarn-install-command-error-no-such-file-or-directory-install)

Let me know if that helps!

**Reply by lucamazzasette** [reply]
Thanks for your support, now I 've another error:

admin@VMRC010:~/carbon$ yarn guided-install
Usage Error: The project in /home/admin/package.json doesn't seem to have been installed - running an install there might help

$ yarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] <scriptName> ...

**Reply by 4upz** [reply]
Apologies, I left out a step in my last message. Try running `yarn install` and then `yarn guided-install`. I've updated my previous message with the correction

**Reply by lucamazzasette** [reply]
I'm very  frustated :(
Error after sudo npx @cloud-carbon-footprint/create-app

internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '/root/.npm/_npx/42903/lib/node_modules/@cloud-carbon-footprint/create-app/node_modules/spawn-sync/postinstall'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! spawn-sync@1.0.15 postinstall: `node postinstall`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the spawn-sync@1.0.15 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2023-05-16T14_44_51_946Z-debug.log
Install for @cloud-carbon-footprint/create-app@latest failed with code 1

I don't understood who install this application.

I would like implement  for testing Carbon footprint for Azure / GCP and on prem but I dont't see results.

Existing a step by step guide to install  ?
Is correct install on Ubuntu?
Which is a best solution (NO AWS()  fot implement this application?

**Reply by 4upz** [reply]
Sorry for the trouble. After further investigation, this looks like a yarn-related issue with the latest version of `create-app` during the app creation step. We will create a separate issue and look into this.

As for now, you should still have the directory created that was named after your app. Change into that folder using `cd [name-of-created-app]` and run the following command to create a lock file for yarn: `touch yarn.lock`

Afterwards, rerun `yarn install` and it should be able to do the installation. After it's successful, you can continue with the `yarn guided-install` step.

**Reply by 4upz** [reply]
As a follow-up to my last message after revisiting some of your steps -- If the app directory didn't finish completing, I would also recommend double-checking your steps to install the [prerequisite tools](https://www.cloudcarbonfootprint.org/docs/getting-started#prerequisites) in your environment as that could also be causing the error you're seeing. For example, on looking at your step to install `yarn` more closely, the installation step for installing yarn classic (v1) which CCF does not support. You should be following [installation steps for yarn v3](https://yarnpkg.com/getting-started/install). Make sure to double-check your node version as well as that will affect which steps you need to take to install yarn.

Unfortunately, we don't have a step-by-step guide for Ubuntu specifically, but installing and running the app on it is possible. The [existing instructions in the documentation](https://www.cloudcarbonfootprint.org/docs/getting-started/#prerequisites) should apply as long as node, yarn, and git (if cloning) is configured correctly.

**Reply by lucamazzasette** [reply]
We  have made progress but not enough.
i'm bloking at point 13 /14 

Now I have the application but it does not allow to yarn start


dmin@UAZWEU01APRC010:~/carbon$ sudo yarn install
➤ YN0000: ┌ Resolution step
➤ YN0061: │ @material-ui/lab@npm:4.0.0-alpha.61 is deprecated: Material UI v4 doesn't receive active development since September 2021. See the guide https://mui.com/material-ui/migration/migration-v4/ to upgrade to v5.
➤ YN0061: │ @types/history@npm:5.0.0 is deprecated: This is a stub types definition. history provides its own type definitions, so you do not need this installed.
➤ YN0061: │ @types/mockdate@npm:3.0.0 is deprecated: This is a stub types definition. mockdate provides its own type definitions, so you do not need this installed.
➤ YN0061: │ @material-ui/core@npm:4.12.4 is deprecated: Material UI v4 doesn't receive active development since September 2021. See the guide https://mui.com/material-ui/migration/migration-v4/ to upgrade to v5.
➤ YN0032: │ fsevents@npm:2.3.2: Implicit dependencies on node-gyp are discouraged
➤ YN0061: │ @material-ui/styles@npm:4.11.5 is deprecated: Material UI v4 doesn't receive active development since September 2021. See the guide https://mui.com/material-ui/migration/migration-v4/ to upgrade to v5.
➤ YN0061: │ read-package-tree@npm:5.3.1 is deprecated: The functionality that this package provided is now in @npmcli/arborist
➤ YN0061: │ uuid@npm:3.4.0 is deprecated: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
➤ YN0061: │ svgo@npm:1.3.2 is deprecated: This SVGO version is no longer supported. Upgrade to v2.x.x.
➤ YN0061: │ rollup-plugin-terser@npm:7.0.2 is deprecated: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
➤ YN0061: │ readdir-scoped-modules@npm:1.1.0 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0061: │ @npmcli/ci-detect@npm:1.4.0 is deprecated: this package has been deprecated, use `ci-info` instead
➤ YN0061: │ stable@npm:0.1.8 is deprecated: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
➤ YN0061: │ querystring@npm:0.2.0 is deprecated: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
➤ YN0061: │ @npmcli/move-file@npm:1.1.2 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0061: │ @npmcli/move-file@npm:2.0.1 is deprecated: This functionality has been moved to @npmcli/fs
➤ YN0061: │ request@npm:2.88.2 is deprecated: request has been deprecated, see https://github.com/request/request/issues/3142
➤ YN0061: │ sourcemap-codec@npm:1.4.8 is deprecated: Please use @jridgewell/sourcemap-codec instead
➤ YN0061: │ har-validator@npm:5.1.5 is deprecated: this library is no longer supported
➤ YN0061: │ w3c-hr-time@npm:1.0.2 is deprecated: Use your platform's native performance.now() and performance.timeOrigin.
comauadmin@UAZWEU01APRC010:~/carbon$ yarn start
yarn start

Internal Error: root-template@workspace:.: This package doesn't seem to be present in your lockfile; run "yarn install" to update the lockfile
    at AF.getCandidates (/home/comauadmin/carbon/.yarn/releases/yarn-3.1.1.cjs:437:4421)
    at pd.getCandidates (/home/comauadmin/carbon/.yarn/releases/yarn-3.1.1.cjs:432:5275)
    at /home/comauadmin/carbon/.yarn/releases/yarn-3.1.1.cjs:441:7674
    at du (/home/comauadmin/carbon/.yarn/releases/yarn-3.1.1.cjs:382:4003)
    at R (/home/comauadmin/carbon/.yarn/releases/yarn-3.1.1.cjs:441:7654)
comauadmin@UAZWEU01APRC010:~/carbon$


The same arror if I try yarn guided-install.

Thnks for your time and support.

**Reply by 4upz** [reply]
Awesome! Glad there's progress. 

This issue sometimes occur when there is a conflicting `yarn.lock` file interrupting the installation. Check your home folder and make sure a `yarn.lock` file wasn't added there by mistake. If so, remove/delete the file. Alternatively, you can also create an empty `yarn.lock` file in your app's root directory using `touch yarn.lock`. You can view more in this Stack Overflow thread: https://stackoverflow.com/questions/63015629/showing-error-in-yarn-start-command-says-this-package-doesnt-seem-to-be-present

Hope that helps!

**Reply by lucamazzasette** [reply]
Finally is working on VM Ubuntu.

Thanks for your time.

**Reply by 4upz** [reply]
No problem, glad to hear the issue is solved!
